packages/plugins/plugin-navtree/src/components/NavTreeItemAction.tsx

```diff
@@ -40,23 +40,20 @@ export const NavTreeItemActionDropdownMenu = ({
 }: NavTreeItemActionMenuProps) => {
   const { t } = useTranslation(NAVTREE_PLUGIN);
   const suppressNextTooltip = useRef<boolean>(false);
-
   return (
-    <DropdownMenu.Root
-      actions={menuActions as MenuAction[]}
-      onAction={onAction}
-      suppressNextTooltip={suppressNextTooltip}
-    >
-      <DropdownMenu.Trigger asChild>
-        <IconButton
-          {...actionButtonProps}
-          icon={icon ?? fallbackIcon}
-          label={toLocalizedString(label, t)}
-          data-testid={testId}
-          suppressNextTooltip={suppressNextTooltip}
-        />
-      </DropdownMenu.Trigger>
-    </DropdownMenu.Root>
+    <MenuProvider onAction={onAction}>
```

Comment:
Why is this separate from the root now? Not using radix-style?

Reply:
Because of nesting… Toolbar itself uses DropdownMenu, and it’s theoretically possible DropdownMenu could render a Toolbar, so there’s no obvious root component, hence the explicit provider.

Reply:
I guess this is another place where I look at the toolbar as the root thing. I might consider naming the `MenuProvider` `Toolbar.Root` instead? It just seems weird to me that the toolbar actions are coming from a hook about menus.

Reply:
I can see the naming is causing a lot of issues, maybe this should not be called toolbar at all? HorizontalStaticMenu? What name would bring better clarity to toolbar’s relatedness with all the other menus?

Reply:
I don't want to get too hung up on it because renaming things is pretty easy. Maybe just stick with what you've got for now and get feedback from Rich/Zaymon on the names as well before deciding if they should be changed (and to what).